include/thousandeyes/futures/detail/InvokerWithSingleThread.h

```diff
@@ -38,7 +38,11 @@ class InvokerWithSingleThread {
                     s->fs.pop();
 
                     lock.unlock();
-                    f();
+                    {
+                        // Ensure f is out of scope before re-acquiring the lock
+                        auto g = std::move(f);
```

Comment:
`std::move` isn't guaranteed to clear out the source object. I think you'll need something like the following:

```cpp
f();
f = function<void()>{};
```

or 
```cpp
{
    auto g = std::exchange(f, {});
    g();
}
```

Reply:
Good point (although very unlikely for non-primitives - or objects with non-trivial dtors). I updated the relevant commit with your suggested change.